flytekit/image_spec/image_spec.py

```diff
@@ -127,11 +126,11 @@ def exist(self) -> bool:
             return False
         except Exception as e:
             tag = calculate_hash_from_image_spec(self)
-            # if docker engine is not running locally
+            # if docker engine is not running locally, use requests to check if the image exists.
             container_registry = DOCKER_HUB
             if self.registry and "/" in self.registry:
                 container_registry = self.registry.split("/")[0]
-            if container_registry == DOCKER_HUB:
+            if container_registry == DOCKER_HUB and "localhost:" not in self.registry:
```

Comment:
don't understand this... if self.registry is localhost something, why is container_registry still DOCKER_HUB?  isn't it more like the if condition above needs to change?

Reply:
updated it